test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+            PDP_PACKETS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            EDP_PACKETS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(EDP_PACKETS_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            DISCOVERY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(DISCOVERY_TOPIC));
+    EXPECT_EQ(discovery_type, statistics_participant->find_type(discovery_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            SAMPLE_DATAS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(SAMPLE_DATAS_TOPIC));
+    EXPECT_EQ(sample_identity_count_type, statistics_participant->find_type(
+            sample_identity_count_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PHYSICAL_DATA_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(physical_data_type, statistics_participant->find_type(physical_data_type.get_type_name()));
+
+    // 5. Enable an enabled statistics DataWriter
+    ret = statistics_participant->enable_statistics_datawriter(
+            SAMPLE_DATAS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+
+    // 6. Invalid topic name
+    ret = statistics_participant->enable_statistics_datawriter(
+            "INVALID_TOPIC", STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription("INVALID_TOPIC"));
+
+    // 7. Disable statistics DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
```

Comment:
As the null type support check is done several times, add a `null_type` variable and compare with it

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3